support for multiple apps

aaronshaf opened the pull request:
- [x] .dat-pub/{application key}.json
- [x] rename endpoint -> /applications/{dat-key}/archives
- [x] resolve dat urls

aaronshaf committed:
support for multiple apps

RangerMauve commented on applications.js:
```diff
@@ -0,0 +1,61 @@
+  })
+
+  // if I load prematurely, it doesn't work
+  sleep(3000).then(loadAllArchives)
```
Should there be an `await` here?

aaronshaf replied:
No, dat location can be immediately returned, even before the hub begins to successfully host it.

RangerMauve replied:
👍 Cool, that makes sense.

RangerMauve commented on utils.js:
```diff
@@ -0,0 +1,38 @@
+    file = await userArchive.readFile(proofLocation)
+  } catch (err) {
+    throw new Error(
+      `Proof file missing: /.well-known/dat-pubs/${applicationPub.url.slice(
```
Should replace this with `Proof file missing: ${proofLocation}` to make it easier to change the location in the future.

aaronshaf replied:
Done

aaronshaf committed:
Overhaul

- Add more schemas
- Add forms for testing API
- Resolve Dat URLs
- Use more Bootstrap on the front

aaronshaf commented:
Worth another look

aaronshaf commented:
Looks like I should use `dat-node` before merging this. https://github.com/beakerbrowser/dat-node/issues/2#issuecomment-408633641

---

Nevermind. I'll do that in a future version.

RangerMauve commented on applications.js:
```diff
@@ -0,0 +1,68 @@
+const { join, basename } = require('path')
+const mkdirp = require('mkdirp')
+
+const APPLICATIONS_DIR = join(__dirname, 'applications')
```
I think this should be configurable by the user.
